datafusion/physical-plan/src/joins/utils.rs

```diff
@@ -563,11 +563,14 @@ fn estimate_inner_join_cardinality(
         .iter()
         .zip(right_stats.column_statistics.iter())
     {
-        // Break if any of statistics bounds are undefined
-        if left_stat.min_value.get_value().is_none()
-            || left_stat.max_value.get_value().is_none()
-            || right_stat.min_value.get_value().is_none()
-            || right_stat.max_value.get_value().is_none()
+        // Break if we don't have enough information to calculate a distinct count
+        // If distinct_count isn't provided directly, we need min and max to be provided
+        if (left_stat.distinct_count.get_value().is_none()
+            && (left_stat.min_value.get_value().is_none()
+                || left_stat.max_value.get_value().is_none()))
+            || (right_stat.distinct_count.get_value().is_none()
+                && (right_stat.min_value.get_value().is_none()
+                    || right_stat.max_value.get_value().is_none()))
```

Comment:
Would this be easier to read?




```suggestion
        let has_enough_info = |stat: &ColumnStatistics| {
            stat.distinct_count.get_value().is_some()
                || (stat.min_value.get_value().is_some()
                    && stat.max_value.get_value().is_some())
        };
        if !has_enough_info(left_stat) || !has_enough_info(right_stat)
```